### PR TITLE
Update cling_build_instructions.md wrt monorepo

### PR DESCRIPTION
**Pull request by Axel-Naumann**

### CODE REVIEW

File: cling/cling_build_instructions.md [hunk 1]
```diff
@@ -23,28 +23,21 @@ Alternatively, you can build manually.
...
 {% endhighlight %}
 
 Then use CMake to configure &amp; build cling:
 
-{% highlight C++ %}
+{% highlight shell %}
 mkdir build
 cd build
-cmake -DCMAKE_INSTALL_PREFIX=[Install Path] -DCMAKE_BUILD_TYPE=[Build configuration, e.g. Release or Debug] ..\src
+cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DLLVM_BUILD_TOOLS=Off -DLLVM_EXTERNAL_PROJECTS=cling -DLLVM_EXTERNAL_CLING_SOURCE_DIR=../cling -DLLVM_ENABLE_PROJECTS=clang -DLLVM_TARGETS_TO_BUILD="host;NVPTX" src/llvm
```
Comment (hahnjo) [inline cling/cling_build_instructions.md:35]:
```suggestion
cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DLLVM_BUILD_TOOLS=Off -DLLVM_EXTERNAL_PROJECTS=cling -DLLVM_EXTERNAL_CLING_SOURCE_DIR=../cling -DLLVM_ENABLE_PROJECTS=clang -DLLVM_TARGETS_TO_BUILD="host;NVPTX" ../src/llvm
```

File: cling/cling_build_instructions.md [hunk 2]
```diff
@@ -23,28 +23,21 @@ Alternatively, you can build manually.
...
 cmake --build .
 cmake --build . --target install
 {% endhighlight %}
 
-### cling-patches?
+### cling patches?
 
 Cling needs a couple of patches on top of clang (but can work with an original build of llvm).
```
Comment (hahnjo) [inline cling/cling_build_instructions.md:42]: This is wrong for LLVM 13, we need patches in LLVM and an original build will not work.

File: cling/cling_build_instructions.md [hunk 3]
```diff
@@ -65,5 +58,9 @@ to package maintainers:
       <a href="https://root.cern/gitweb/?p=clang.git;a=commit;h=27d2d5b247e83f3ac5d6e9f36d25d168a6afc97a">27d2d5</a>
       (aka <a href="https://root.cern/gitweb/?p=clang.git;a=shortlog;h=refs/tags/cling-patches-rrelease_50">cling-patches-rrelease_50</a>)
       </td>
+      <td>0.7</td>
+      <td>9.0 release series</td>
+      <td>
+      </td>
```
Comment (hahnjo) [inline cling/cling_build_instructions.md:64]: Did you mean to include this? In a separate row? (there are only three columns)

**Comment by hahnjo** [comment]
I just created a branch [`cling-latest`](https://github.com/root-project/llvm-project/tree/cling-latest) pointing to `cling-llvm13` for the moment. Maybe we should add a note what users should expect from this branch, something like
> The `cling-latest` branch tracks the evolution of patches in our fork of the LLVM monorepo. Developers should expect this reference to move when new changes are added or modified, or jump to a new version of LLVM in sync with `cling.git`. If you are not planning to modify the LLVM sources locally, we recommend fetching changes via `git fetch` and then resetting the branch to match the remote: `git reset --hard origin/cling-latest`.

**Comment by silver-ymz** [comment]
Hi, there seems to be a problem with the installation process. I installed cling following the build instructions. After deleting the build directory, `cling` gives me an error.

```shell
> pwd
/home/ymz/tmp
... build commands
> mv build build-a
> cling
ERROR in cling::CIFactory::createCI():
  resource directory /home/ymz/tmp/build/lib/clang/13.0.0 not found!

****************** CLING ******************
* Type C++ code and press enter to run it *
*             Type .q to exit             *
*******************************************
[cling]$ .q
```

**Comment by gnrv** [comment]
I just wanted to say these updated instructions work great, tried it both in WSL Ubuntu 22.04 with GCC and Win11 with VSC 17 2022. But they sure were hard to find - it would be super helpful if the instructions on the web were updated, and perhaps the cling repo itself included some sort of pointers in the right direction.

**Comment by Axel-Naumann** [comment]
Thanks for the feedback. Let's get that fixed, starting with merging this!

**Pull request merged**